### PR TITLE
Swift: subsume IterableDeclContext into Decl

### PR DESCRIPTION
**Pull request by d10c**
- Swift: subsume IterableDeclContext into Decl [hand-written]
- Swift: subsume IterableDeclContext into Decl [codegen'd]
- Swift: subsume IterableDeclContext into Decl [tests]
- Swift: up/downgrade scripts

### CODE REVIEW

File: swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: revert removal of IterableDeclContext
+compatibility: partial
```
Comment (geoffw0) [inline swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties:2]: Is it fair to assume that any `Decl` with members is an `IterableContextDecl` after the downgrade?  If so, do we actually lose anything in the downgrade?

**Review by geoffw0 — COMMENTED** [review]
Looks good to me.  A few very minor suggestions.

I did have a small concern that the dbscheme changes could have a negative effect on analysis performance, because what can and can't have members is no longer so tightly constrained.  But the numbers from the DCA runs appear to contradict this concern.  Also the latest DCA run LGTM (apart from the source archive hashes / temporary files issue we discussed earlier).

How have the upgrade/downgrade scripts been tested?

**Comment by geoffw0** [comment]
Let me know if there's anything I can do to help unblock you here, please.

File: swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties (hunk 1 above)
Comment (d10c) [reply swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties]: Yes, the downgrade script specifies that only `@nominal_type_decl or @extension_decl` (i.e. the old `@iterable_context_decl`) will revert back to being an IterableContextDecl with members. What we could lose are any future Decl subtypes that override the `getImmediateMember()` predicate, for instance.

I've been having trouble testing the db upgrade/downgrade script though; against the same baseline (upstream/main more or less), I can only successfully run the upgrade script, not the downgrade script. I've been following the steps from [here](https://github.com/github/codeql/blob/main/docs/prepare-db-upgrade.md).

Comment (geoffw0) [reply swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties]: Yes, I have a similar situation in https://github.com/github/codeql/pull/12001 - I can test the upgrade script but I don't know how to test the downgrade script.  I'll ask around if anyone knows how to do this...

Comment (d10c) [reply swift/downgrades/264374a5a3d4445024add4c63e657b053f6165e8/upgrade.properties]: It should run fine now 🤞🏻 Managed to test the upgrade/downgrade scripts.

**Review by geoffw0 — APPROVED** [review]
All concerns addressed.  Looks good to me.

**Pull request merged**